ee/clickhouse/models/property.py

```diff
@@ -385,14 +400,34 @@ def get_property_values_for_key(key: str, team: Team, value: Optional[str] = Non
     parsed_date_from = "AND timestamp >= '{}'".format(relative_date_parse("-7d").strftime("%Y-%m-%d 00:00:00"))
     parsed_date_to = "AND timestamp <= '{}'".format(timezone.now().strftime("%Y-%m-%d 23:59:59"))
 
+    property_string_expr, _ = get_property_string_expr("events", key, "%(key)s", "properties", True)
```

Comment:
If you're refactoring this, ideally also pull out a query object here and move this outside the model file.

Reply:
Is that to move this knowledge

`
if "FROM JSONExtractRaw" in property_string_expr:
        # only include rows which have the desired key in the properties blob
        existence_check = "AND JSONHas(properties, %(key)s)"
    else:
        # `property_string_expr` is either a table level column or a materialized column
        # so only include rows where the value is present
        existence_check = f"AND isNotNull(NULLIF({property_string_expr}, ''))"
`

out of the model?

Reply:
or take the whole `get_property_string_expr` lump out into its own object

Reply:
Sorry - meant this whole method `get_property_values_for_key` which is used in a single query somewhere shouldn't live in this file.